cvpaperchallenge/Ascender
# #95/chore/check new linter ruff

## Issue URL

Close: #95 

## Change overview

- Add the new linter `ruff` written in Rust, and remove `flake8` and `isort` from dev dependencies
- Update the python libraries listed on dev dependencies

[Add changes below on 2024/04/27]
- Replace `black` with [Ruff formatter](https://docs.astral.sh/ruff/formatter/) (released as of v0.1.2)
- Add the recommended Ruff linter rules compatible with Ruff formatter [[link](https://docs.astral.sh/ruff/formatter/#conflicting-lint-rules)]
- Remove `.flake8` file
- Describe how to loosen the rules of Ruff in `README.md`
- Update `README.md` representation

## How to test

- ~~Make sure to work `make ruff`, `make ruff-check` under the core container~~
- Make sure to run `make format` with no warnings under the core container

## Note for reviewers

N/A
